src/instance/ListInstance.ts

```diff
@@ -55,9 +55,19 @@ export default class ListInstance implements Searchable {
     this.children.set(keys, newChild);
   }
 
-  public toJSON(camelCase = false, convert = true): { [name: string]: ListJSON } {
+  public toJSON(
+    camelCase = false,
+    convert = true,
+    shouldSkip = (ins: Instance) => false
+  ): { [name: string]: ListJSON } {
+    const value = [];
+    for (const child of this.children.values()) {
+      if (!shouldSkip(child)) {
```

Comment:
What is instead of defaulting the `shouldSkip = (ins: Instance) => false` function all the way down, you make it truly optional.  Then you could do a performance optimization here where you check first if should skip is even defined, if it is do the loop you have here, if it isn't just blindly push everything onto the result.  That way you can avoid wasted function calls and conditionals (granted a little more verbose code-wise, but this will get called a lot)

Reply:
I'm not sure if there are other things being considered here... Could this also be done slightly more functionally? 
```
return {
   [this.model.getName(camelCase)]: this.children.values()
       .filter(x => !shouldSkip || shouldSkip(x))
       .map(x => x.toJSON(camelCase, convert))
};
```
This should handle both using `shouldSkip` as an optional and be slightly more functional.

Reply:
We talked a bit about doing it the way he has it for performance, v8 may do shortcut fusion of filter & map, but we wanted to be sure

Reply:
Figured that was the reasoning. It looks like your getting a `50%` speed boost even in pretty small data sizes ~ `100,000`